src/test-run/commands/actions.js

```diff
@@ -299,70 +260,63 @@ export class PressKeyCommand extends Assignable {
     }
 }
 
-export class NavigateToCommand extends Assignable {
-    constructor (obj) {
-        super(obj);
+export class NavigateToCommand extends CommandBase {
+    constructor (obj, testRun) {
+        super(obj, testRun);
 
-        this.type          = TYPE.navigateTo;
-        this.url           = null;
-        this.stateSnapshot = null;
-
-        this._assignFrom(obj, true);
+        this.type = TYPE.navigateTo;
     }
 
     _getAssignableProperties () {
         return [
             { name: 'url', type: urlArgument, required: true },
-            { name: 'stateSnapshot', type: nullableStringArgument }
+            { name: 'stateSnapshot', type: nullableStringArgument, defaultValue: null }
         ];
     }
 }
 
-export class SetFilesToUploadCommand extends Assignable {
-    constructor (obj) {
-        super(obj);
+export class SetFilesToUploadCommand extends CommandBase {
+    constructor (obj, testRun) {
+        super(obj, testRun);
 
         this.type = TYPE.setFilesToUpload;
-
-        this.selector = null;
-        this.filePath = '';
-
-        this._assignFrom(obj, true);
     }
 
     _getAssignableProperties () {
         return [
-            { name: 'selector', init: (name, val) => initSelector(name, val, true), required: true },
+            { name: 'selector', init: (name, val, options) => {
```

Comment:
rename please `options` -> `initOptions` (here and below), to avoid mix it up with `actionOptions`